osu.Game/Skinning/RulesetSkinProvidingContainer.cs

```diff
@@ -60,6 +73,9 @@ protected override void OnSourceChanged()
 
         protected virtual void UpdateSkins()
         {
+            foreach (var resourcesSkin in SkinSources.OfType<ResourcesSkin>())
+                resourcesSkin.Dispose();
```

Comment:
Why? Are you guarding against a ruleset changing during runtime or something..?

Reply:
Because it's recreated every skin, but since you're proposing not to do that, alright.

Reply:
You would be flushing every resource for no reason, which will potentially be a performance concern for rulesets which have already loaded considerable resources. There's a reason we don't do this for other skins.